linera-execution/src/block_executor.rs

```diff
@@ -0,0 +1,328 @@
+// Copyright (c) Zefchain Labs, Inc.
```

Comment:
It'd be awesome if we could have a rustdoc here explaining what's the idea for `BlockExecutor`.